GHzDACs/jumpTable.py

```diff
@@ -4,182 +4,234 @@
+        return '\n'.join([f_str, t_str, op_str])
+
+    def as_bytes(self):
+        data = np.zeros(8, dtype='<u1')
+        data[0:3] = littleEndian(self.from_addr, 3)
+        data[3:6] = littleEndian(self.to_addr, 3)
+        data[6:8] = littleEndian(self.operation.as_bytes(), 2)
         return data
 
+
 # Operations (ie op codes)
-        
+
 class Operation(object):
     """A Super class for all possible jump table operations"""
-    def getJumpIndex(self):
-        return self._jumpIndex
-    def setJumpIndex(self, idx):
-        if JUMP_INDEX_MIN < idx and idx < JUMP_INDEX_MAX:
-            self._jumpIndex = idx
+    NAME = 'INVALID'  # subclasses must override
+
+    @property
+    def jump_index(self):
+        return self._jump_index
+
+    @jump_index.setter
+    def jump_index(self, idx):
+        if JUMP_INDEX_MIN <= idx <= JUMP_INDEX_MAX:
+            self._jump_index = idx
         else:
-            raise RuntimeError("Must have %s < jump index < %s"%(JUMP_INDEX_MIN, JUMP_INDEX_MAX))
-    jumpIndex = property(getJumpIndex, setJumpIndex)
-    
+            raise ValueError("Must have {0} < jump index < {1}, not {2}".format(JUMP_INDEX_MIN, JUMP_INDEX_MAX, idx))
+
     def __str__(self):
-        raise NotImplementedError
-    
-    def asBytes(self):
+        raise NotImplementedError()
+
+    def as_bytes(self):
         """Override in subclass"""
-        raise NotImplementedError
+        raise NotImplementedError()
+
 
 class IDLE(Operation):
-    """Wraps the IDLE jump table op code"""
-    
     NAME = "IDLE"
-    
+
     def __init__(self, cycles):
-        self._cycles = cycles
-    
-    def getJumpIndex(self):
-        raise RuntimeError('IDLE does not support jump table indexing')
-    def setJumpIndex(self, idx):
-        raise RuntimeError('IDLE does not support jump table indexing')
-    
-    def getCycles(self):
+        self._cycles = 0
+        self.cycles = cycles - 1  # account for FPGA firmware timing offset
+
+    @property
+    def cycles(self):
         return self._cycles
-    def setCycles(self, cycles):
-        if IDLE_MIN_CYCLES < cycles and cycles < IDLE_MAX_CYCLES:
+
+    @cycles.setter
+    def cycles(self, cycles):
+        if IDLE_MIN_CYCLES < cycles < IDLE_MAX_CYCLES:
             self._cycles = cycles
         else:
-            raise RuntimeError('Number of idle cycles must satisfy: %s < cycles < %s'%(IDLE_MIN_CYCLES,IDLE_MAX_CYCLES))
-    cycles = property(getCycles, setCycles)
-    
+            raise ValueError(
+                'Number of idle cycles must satisfy: {0} < cycles < {1}, '
+                'not {2}'.format(IDLE_MIN_CYCLES, IDLE_MAX_CYCLES, cycles)
+            )
+
     def __str__(self):
-        return "%s %d cycles"%(self.NAME, self.cycles)
-    
-    def asBytes(self):
-        return self._cycles<<1
-    
-class CHECK(Operation):
+        return "%s %d cycles" % (self.NAME, self.cycles)
 
+    def as_bytes(self):
+        return self._cycles << 1
```

Comment:
It seems to me that `as_bytes` should return bytes like `JumpEntry` does, not an integer.